src/PhpGenerator/Parameter.php

```diff
@@ -39,16 +39,21 @@ public static function from(\ReflectionParameter $from)
 		$param = new static;
 		$param->name = $from->getName();
 		$param->reference = $from->isPassedByReference();
-		if ($from->isArray() || $from->isCallable()) {
-			$param->typeHint = $from->isArray() ? 'array' : 'callable';
+		if (PHP_VERSION_ID >= 70000 && $from->hasType()) {
+			$type = $from->getType();
+			$param->typeHint = $type->isBuiltin() ? (string) $type : '\\' . $type;
 		} else {
```

Comment:
I have a feeling this could be done in a better way - maybe `else if (PHP_VERSION_ID < 70000)`? Because if the code doesn't have typehint, PHP 7.0 would try to execute the else branch pointlessly?